client/data/marketplace/use-marketplace-reviews.ts

```diff
@@ -0,0 +1,134 @@
+import {
+	useQuery,
+	UseQueryResult,
+	UseQueryOptions,
+	QueryKey,
+	useMutation,
+	useQueryClient,
+} from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import { BASE_STALE_TIME } from 'calypso/state/initial-state';
+
+const reviewsApiBase = '/sites/marketplace.wordpress.com/comments';
+const reviewsApiNamespace = 'wp/v2';
+const queryKeyBase: QueryKey = [ 'marketplace-reviews' ];
+
+type ProductType = 'plugin' | 'theme';
+
+type ProductProps = {
+	productType: ProductType;
+	pluginSlug: string;
+};
+
+type MarketplaceReviewBody = {
+	content: string;
+	rating: number;
+} & ProductProps;
+
+type UpdateMarketplaceReviewProps = {
+	reviewId: number;
+} & MarketplaceReviewBody;
+
+type DeleteMarketplaceReviewProps = {
+	reviewId: number;
+} & ProductProps;
+
+const fetchMarketplaceReviews = ( productType: ProductType, productSlug: string ) => {
+	return wpcom.req.get(
+		{
+			path: reviewsApiBase,
+			apiNamespace: reviewsApiNamespace,
+		},
+		{
+			product_type: productType,
+			product_slug: productSlug,
+		}
+	);
+};
+
+const createReview = ( { productType, pluginSlug, content, rating }: MarketplaceReviewBody ) => {
+	return wpcom.req.post(
+		reviewsApiBase,
+		{
+			apiNamespace: reviewsApiNamespace,
+		},
+		{
+			product_type: productType,
+			product_slug: pluginSlug,
+			content,
+			meta: { wpcom_marketplace_rating: rating },
+		}
+	);
+};
+
+const updateReview = ( {
+	reviewId,
+	productType,
+	pluginSlug,
+	content,
+	rating,
+}: UpdateMarketplaceReviewProps ) => {
+	return wpcom.req.post(
+		`${ reviewsApiBase }/${ reviewId }`,
+		{
+			apiNamespace: reviewsApiNamespace,
+		},
+		{
+			product_type: productType,
+			product_slug: pluginSlug,
+			content,
+			meta: { wpcom_marketplace_rating: rating },
+		}
+	);
+};
+
+const deleteReview = ( { reviewId }: DeleteMarketplaceReviewProps ) => {
+	return wpcom.req.del( `${ reviewsApiBase }/${ reviewId }`, {
+		apiNamespace: reviewsApiNamespace,
+	} );
+};
+
+export const useMarketplaceReviews = (
+	{ productType, pluginSlug }: ProductProps,
+	{ enabled = true, staleTime = BASE_STALE_TIME, refetchOnMount = true }: UseQueryOptions = {}
```

Comment:
If you don't want to expose the whole API surface for `useQuery` I'd generally advise to narrow it down to just the properties you know you're going to use – you already did that, however, `UseQueryOptions` used like that is not the correct type to use. Some alternatives:

```ts
type MarketplaceReviewsQueryOptions = Pick<
	UseQueryOptions,
	'enabled' | 'staleTime' | 'refetchOnMount'
>;
```

If you type your fetch function like I suggested:

```ts
type MarketplaceReviewsQueryOptions = Pick<
	UseQueryOptions< MarketplaceReviewsResponse >,
	'enabled' | 'staleTime' | 'refetchOnMount'
>;
```

As soon as you expose any properties which make use of `data` using the generic is recommended because otherwise it won't receive the correct `data` type (`refetchOnMount` accepts a function which is given the query which also contains `data`).

In any case what you should do because it will type error in v5 is omitting `queryKey` from your options type when using `UseQueryOptions` like that (not needed when you do one of the above):

```ts
const options: Omit< UseQueryOptions, 'queryKey' > = { /* ... */ }
// or
const options: Omit< UseQueryOptions< MarketplaceReviewsResponse >, 'queryKey' > = { /* .. */ }
```